lib/lib-storage/src/Upload.ts

```diff
@@ -364,6 +369,12 @@ export class Upload extends EventEmitter {
 
     let result;
     if (this.isMultiPart) {
+      if (this.totalBytes && this.uploadedParts.length !== this.expectedPartsCount) {
```

Comment:
`totalBytes` can be zero and `expectedPartsCount` would be 1 in that case.

This should instead be
```js
const { expectedPartsCount, uploadedParts } = this;

if (expectedPartsCount !== undefined 
    && uploadedParts.length !== expectedPartsCount) {
```